src/main/java/Calculator.java

```diff
@@ -1,23 +1,39 @@
 import java.util.List;
 
 public class Calculator {
+    private static Calculator instance;
```

Comment:
Singleton rarely has good use cases. this is because it indirectly violates single responsibility principle (class is available everywhere) . You should only use singleton if you change state of the class in multiple (most of) different classes. In this case your `Calculator` is stateless, so using Singleton makes no sense.